Make this package easier to run

nylen opened the pull request:
This PR updates the readme to add full installation and usage instructions at the top of the readme. It adds a `bin/cpcs` script which is both an easier way to run this ruleset and a next step towards automating this process for plugins. This PR also makes a couple of changes to the `phpcs.xml` rules:

- Disable the `Squiz.PHP.EmbeddedPhp.ContentBeforeEnd` formatting rule which showed up for me while I was playing with this project.
- Respect `phpcs:ignore` annotations, and document in the readme how to show all warnings/errors including those that are ignored. Specific `phpcs:ignore` comments accompanied by explanations of why a given rule is being ignored should be accepted as a way to pass the coding standards checks.

It's better to specify options _at runtime_ rather than requiring users to edit the `phpcs.xml` directly. For the `text_domain` setting, this can be done via `--runtime-set text_domain plugin-textdomain`: https://github.com/WordPress/WordPress-Coding-Standards/blob/7da1894633f168fe244afc6de00d141f27517b62/WordPress/Sniffs/WP/I18nSniff.php#L193-L197

The `minimum_supported_wp_version` setting should have similar treatment, but I think this should be saved for a separate PR: https://github.com/WordPress/WordPress-Coding-Standards/blob/dc2f21771cb2b5336a7e6bb6616abcdfa691d7de/WordPress/Helpers/MinimumWPVersionTrait.php#L60

nylen commented on bin/cpcs:
```diff
@@ -0,0 +1,24 @@
+
+if [ -z "$1" ] || [ -z "$CPCS_TEXTDOMAIN" ]; then
+	echo "Usage: CPCS_TEXTDOMAIN=plugin-textdomain $0 plugin-directory-name [other-wpcs-args]"
+	echo "(or use . for current directory)"
```
```suggestion
	echo "(to check the current directory, use . for plugin-directory-name)"
```

TukuToi reviewed (changes requested):
This all looks good to me, the only I want to follow persistently is the order of rules in the file, so this 
```
<exclude name="Squiz.PHP.DisallowMultipleAssignments.FoundInControlStructure"/>
<exclude name="Squiz.PHP.CommentedOutCode.Found"/>
<exclude name="Squiz.Strings.ConcatenationSpacing.PaddingFound"/>
<exclude name="Squiz.PHP.EmbeddedPhp.ContentBeforeEnd"/>
```

Would have to become this
```
<exclude name="Squiz.PHP.CommentedOutCode.Found"/>
<exclude name="Squiz.PHP.DisallowMultipleAssignments.FoundInControlStructure"/>
<exclude name="Squiz.PHP.EmbeddedPhp.ContentBeforeEnd"/>
<exclude name="Squiz.Strings.ConcatenationSpacing.PaddingFound"/>
```

(I am aware that the current file probably still has some non-alphabetically ordered rules, but the intention is that we'd order the rules alphabetically by first letter, for each ruleset, starting at the first word and then second, third, etc)
It stays more structured and easy to find things

nylen commented:
Ok, no problem. I fixed the rule order in the whole file. It would be a good idea to set up a GitHub Actions job so that it stays that way, but that is out of scope for this PR.

I also fixed a bunch of whitespace issues (indentation with a mix of tabs and spaces, spaces after the end of lines). I am all for keeping spacing consistent in my own code, but we can't enforce this kind of thing on everyone as it is a purely cosmetic issue.
